engine/collection/epochmgr/engine.go

```diff
@@ -17,8 +18,15 @@ import (
 	"github.com/onflow/flow-go/state/protocol/events"
 )
 
+// DefaultStartupTimeout is the default time we wait when starting epoch
+// components before giving up.
 const DefaultStartupTimeout = 30 * time.Second
 
+// ErrUnstakedForEpoch is returned when we attempt to create epoch components
+// for an epoch in which we are not staked. This is the case for epochs during
+// which this node is joining or leaving the network.
+var ErrUnstakedForEpoch = fmt.Errorf("we are not a staked node in the epoch")
```

Comment:
we might be ambiguous especially if the logs are collected or debugged by partner nodes, I think including node_id and epoch number might be more informative.

Reply:
Added to the error message here https://github.com/onflow/flow-go/pull/238/commits/e5f47bb44a37fb32529a2d07be1dca5482a8d88c